### DISCUSSION TITLE
[Help]: selectedScrollSnap() doesn't return indexes for some snaps

### DISCUSSION THREAD

**Discussion by Wakinki** [davidjerleke/embla-carousel#1023] [Help] [answered]
### Summary

I have a carousell with slides the same as my viewport and I want to get the index of the selected slide. So selectedScrollSnap should work on this and return the index of the selected slide. But it does it only for the first second and last slides depending on what direction you were sliding for some reason. I Don't know what causes this because I use the same function on another bit different carousell and it works flawlessly. Here is a codesandbox example: 

https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-kmg79g?file=%2Fsrc%2Fjs%2FEmblaCarousel.tsx%3A44%2C41

### If applicable, which variants of Embla Carousel are relevant to this question?

- [X] embla-carousel **(Core)**
- [X] embla-carousel-react
- [ ] embla-carousel-vue
- [ ] embla-carousel-svelte
- [ ] embla-carousel-autoplay
- [ ] embla-carousel-auto-scroll
- [ ] embla-carousel-solid
- [ ] embla-carousel-auto-height
- [ ] embla-carousel-class-names
- [ ] embla-carousel-fade
- [ ] embla-carousel-docs **(Documentation)**
- [ ] embla-carousel-docs **(Generator)**

### Additional information

I am using "embla-carousel-react": "^8.1.7", and the only difference I can see between the working and the non working carousell is that the non working one has prev and next buttons and some different styles

### CodeSandbox example

https://codesandbox.io/p/sandbox/embla-carousel-generator-react-forked-kmg79g?file=%2Fsrc%2Fjs%2FEmblaCarousel.tsx%3A44%2C41

**Comment by davidjerleke** [comment, marked as answer]
Hi @Wakinki,

Thanks for your question. In your code, `selectedScrollSnap` will only update when your component re-renders. What you want to do is the same thing as the dot button file, in one of the examples on the documentation page that has dots. For example, the [default example](https://www.embla-carousel.com/examples/predefined/#default):

- Generate the sandbox.
- Open the `EmblaCarouselDotButton` file.
- Follow how the `selectedIndex` is used.

**Reply by Wakinki** [reply]
Ah, yes I see I have allready implemented something similar on my other carousell and thats why it worked. 
Thanks for the fast help and keep up the good work 😄.
